Moved content picker into a popover

kugiigi opened the pull request:
Originally from #336 

- Put the content picker when opening downloads into a popup for a more "convergent" design.
- Also moved here the open in browser action buttons

kugiigi committed:
moved content picker to a popover

UniversalSuperBox reviewed (changes requested):
This appears to introduce a warning when the content picker is opened by tapping on a download:

```file:///usr/lib/arm-linux-gnueabihf/qt5/qml/Ubuntu/Content/ContentPeerPicker13.qml:196: TypeError: Type error```

We should investigate why this happens, and if possible fix it, before merging.

kugiigi committed:
updated copyright texts and enabled click logic in downloads dialog

kugiigi commented:
> This appears to introduce a warning when the content picker is opened by tapping on a download:
> 
> `file:///usr/lib/arm-linux-gnueabihf/qt5/qml/Ubuntu/Content/ContentPeerPicker13.qml:196: TypeError: Type error`
> 
> We should investigate why this happens, and if possible fix it, before merging.

I can't find exactly what's causing the error so I decided to just fix those annoying `TypeError` errors in content hub :grinning:  
ubports/content-hub#17

I've also rebased it and fixed the conflicts. I also enabled the commented out codes for opening files from the downloads dialog.
I hope it's fine otherwise, sorry :sweat_smile:

UniversalSuperBox commented on src/app/DownloadsDialog.qml:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2015-2016 Canonical Ltd.
+ * Copyright 2021 UBports Foundation
```
Whoops... In this case, both copyright statements would be kept. The code was under copyright by Canonical, and now we've added some stake with our changes.

kugiigi replied:
No, that file was created in #415 
i actually noticed that there are a few others that we missed in that PR :sweat_smile: 
I'm planning to correct them in a separate PR but I just corrected this one here since I modified it anyway.

But this is definitely a lesson learned for me moving forward - don't just copy paste copyright texts :grinning:

kugiigi committed:
Replaced UITK popup with QQC2

kugiigi commented:
I changed the dialog from UITK to QQC2 since it's more flexible with placement and sizing.
And it also fixes the issue with UITK popup's visibility and avoid the crash issue on specific scaling (#311)

This also sets the qt version to 5.12 because it was needed to set the dimming background of the popup.

UniversalSuperBox commented:
Here is a preview of this change at different scale levels

OnePlus One portrait
![image](https://user-images.githubusercontent.com/21966173/118688474-b520c580-b7cb-11eb-819f-ed4a502e8063.png)

OnePlus One landscape
![image](https://user-images.githubusercontent.com/21966173/118688557-c7026880-b7cb-11eb-96b4-fb96789d2c91.png)

Nexus 7 portrait
![image](https://user-images.githubusercontent.com/21966173/118688588-cf5aa380-b7cb-11eb-9b54-d9c443326ceb.png)

Nexus 7 landscape
![image](https://user-images.githubusercontent.com/21966173/118688611-d4b7ee00-b7cb-11eb-8e3e-bf9731297b9a.png)

Nexus 7 windowed (default scaling)
![image](https://user-images.githubusercontent.com/21966173/118688638-daadcf00-b7cb-11eb-8f9c-2bb67910f326.png)

Nexus 7 windowed (small scaling)
![image](https://user-images.githubusercontent.com/21966173/118688669-e00b1980-b7cb-11eb-9614-a0af612a3a1d.png)

cibersheep commented:
Oh. This is a improvement.
Is nice to have the filename in the header and that the header takes less space.

I see the dialog on portrait is anchored to the bottom. They should center vertically as well.

Question (out of curiosity): Why did you choose a Dialog instead of a Page?
Question (might be off topic): What's the difference between the header actions folder and external link? The file is open in the browser whatever I choose.

kugiigi commented:
> I see the dialog on portrait is anchored to the bottom. They should center vertically as well.

I can't remember exactly the logic when it anchors at the bottom but I think it only happens at certain height and width and not just when in portrait. Otherwise, the dialog is centered (see last picture). I designed it like that so that it's easily reachable. It's a bit similar with android's dialog when opening a file or an intent.

> Question (out of curiosity): Why did you choose a Dialog instead of a Page?

A dialog can be positioned more freely and can be designed with convergence in mind. The content picker doesn't need to be always displayed full especially on large screens.

> Question (might be off topic): What's the difference between the header actions folder and external link? The file is open in the browser whatever I choose

These are existing actions that were not implemented on this PR. In my understanding, one opens the local copy of the file in the browser while the other opens the original dowload link of the file.

cibersheep commented:
> > I see the dialog on portrait is anchored to the bottom. They should center vertically as well.
> 
> I can't remember exactly the logic when it anchors at the bottom but I think it only happens at certain height and width and not just when in portrait. Otherwise, the dialog is centered (see last picture). I designed it like that so that it's easily reachable. It's a bit similar with android's dialog when opening a file or an intent.

Dialogs in UT are centered horizontally and vertically. Moprh being a core app should follow the same UI design.

Thanks for the other answers :)

kugiigi commented:
Then maybe we should change that design 😝
I'll change this when I get the time. Unforetunately, I'm not sure when that would be 😅

kugiigi committed:
Change sizing behavior of content export popup

kugiigi commented:
I've changed the behavior of the popup. It's now full screen for most phones, not anchored at the bottom.

cibersheep commented:
There is a dancing sheep somewhere :)

Fuseteam commented:
what is blocking this PR :3
